proto/job.proto

```diff
@@ -546,7 +546,9 @@ message Job {
     string group = 7;
     string facility = 8;
     string os = 9;
-    int32 uid = 10;
+    oneof uid_optional {
+        int32 uid = 10;
```

Comment:
Isn't there an `optional` keyword you can add as a prefix to do this?

Reply:
Not in version 3 of Protobuf. In v3 all fields are optional, and there’s no way to distinguish between "not provided and defaulted" or "provided with default value". Using `oneof` lets us distinguish the not-provided case.